source/configuration_manual/event_filter.rst

```diff
@@ -48,25 +48,35 @@ The key-value comparisons are of the form: ``<key> <operator> <value>``
 
 Where the key is one of:
 
-1. ``event``
-2. ``category``
-3. ``source_location``
-4. a field name
+#. ``event``
+#. ``category``
+#. ``source_location``
+#. a field name
 
 The operator is one of:
 
-1. ``=``
-2. ``>``
-3. ``<``
-4. ``>=``
-5. ``<=``
+#. ``=``
+#. ``>``
+#. ``<``
+#. ``>=``
+#. ``<=``
 
 And the value is either:
 
-1. a single word token, or
-2. a quoted string
+#. a single word token, or
+#. a quoted string
 
 The value may contain wildcards if the comparison operator is ``=``.
+The value comparison is case-insensitive, but the key is case-sensitive.
+
+There are some limitations on which operators work with what field types:
+
+* string: Only the ``=`` operator is supported.
+* number: All operators are supported.
+* timestamp: No operators are supported.
+* a list of strings: Only the ``=`` operator is supported.
```

Comment:
Should we add a special operator for this? Something like "IS"? "HAS"? "CONTAINS"?  For example:

```
event=abc AND reason IS "foo:bar"
event=abc AND reason HAS "foo:bar"
event=abc AND reason CONTAINS "foo:bar"
```

It's is pretty simple to add the new operator and it'd make it more obvious that we're dealing with something different.

Hmm, I just realized something about the list of strings and (ab)using the '=' operator.  It'll make optimizations harder (or possibly impossible).  Specifically, consider the following expression:

```
abc="foo" AND abc="bar"
```

Without list-of-strings, this is obviously always false and in the future we should take advantage of that (and even warn on startup that filter never matches anything).

With list-of-strings, this can be sometimes true and sometimes false and there is no way to know ahead of time - we have no idea if we are dealing with a string comparison of a membership check until we are evaluating the filter with a specific event.  This prevents the always true/always false sort of expression simplifications.

So, I think we definitely want to have a separate operator for this membership check.

Reply:
The same issue already exists for category..